components/autocomplete/index.js

```diff
@@ -410,37 +464,38 @@ export class Autocomplete extends Component {
 					focusOnOpen={ false }
 					onClose={ this.reset }
 					position="top right"
-					className={ classes }
+					className="components-autocomplete__popover"
 					getAnchorRect={ this.getWordRect }
 				>
-					<ul
+					<div
 						id={ listBoxId }
 						role="listbox"
 						className="components-autocomplete__results"
 					>
-						{ map( filteredOptions, ( option, index ) => (
-							<li
+						{ isExpanded && map( filteredOptions, ( option, index ) => (
+							<Button
 								key={ option.key }
-								id={ `components-autocomplete-item-${ instanceId }-${ index }` }
+								id={ `components-autocomplete-item-${ instanceId }-${ option.key }` }
 								role="option"
 								aria-selected={ index === selectedIndex }
-								className={ classnames( 'components-autocomplete__result', {
+								className={ classnames( 'components-autocomplete__result', className, {
 									'is-selected': index === selectedIndex,
 								} ) }
+								onClick={ () => this.select( option ) }
 							>
-								<Button onClick={ () => this.select( option ) }>
-									{ option.label }
-								</Button>
-							</li>
+								{ option.label }
+							</Button>
 						) ) }
-					</ul>
+					</div>
 				</Popover>
 			</div>
 		);
+		/* eslint-enable jsx-a11y/no-static-element-interactions, jsx-a11y/onclick-has-role, jsx-a11y/click-events-have-key-events */
 	}
 }
 
 export default flowRight( [
+	withSpokenMessages,
 	withInstanceId,
-	withFocusOutside,
+	withFocusOutside, // this MUST be the innermost HOC as it calls handleFocusOutside
```

Comment:
Alternatively we could consider incorporating something like `hoist-non-react-statics` so we don't depend on a particular order for composing higher-order components.

https://www.npmjs.com/package/hoist-non-react-statics